### DISCUSSION TITLE
Welcome to Disk-Utilities Discussions!

### DISCUSSION THREAD

**Discussion by keirf** [keirf/disk-utilities#84] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by sethm** [comment]
Hello, my name is Seth, I'm a developer from Washington State in the United States! I recently bought a Greaseweazle v4 with the intention of archiving some 3B2 floppy disks. I'd like to add support for the 3B2 format to disk-utilities, and I'm curious if there's a "Getting Started" or "Hacking" guide somewhere for developers who want to add support for a new format?

The disks are a little unusual, but easily readable on a standard PC floppy drive like a TEAC FD-55GFR or FD-55FV. They are 5.25" double density with 80 tracks instead of 40 tracks (96tpi instead of 48tpi). They store 720KB with 250kbps MFM encoding, 9 sectors per track, 2 heads, with a 3:1 interleave. I don't think it would be too difficult to add this format and I'd be happy to do the work.

**Comment by keirf** [comment]
Probably not too hard

1. Add a new track_handler at the bottom of ibm_img.c
2. Add a new format name in track_types.h
3. Add a new format descriptor in disk-analyse/formats

Probably the main difficulty will be that ibm_img.c shamefully does not support interleave (or skew). So that needs adding as a track_handler parameter, and supporting within ibm_img.c

**Comment by sethm** [comment]
Well, actually I feel quite silly because it looks like the `atari_st_720kb` format is exactly the same as the 3B2 format, perhaps except for the interleave. I just tried a conversion of an SCP file and it worked like a charm, so I'm hopeful that this might actually be good enough for my needs! If not, at least now I have a good place to start looking at modifications.

Thank you!
